### ISSUE TITLE
Datetime widget issues

### ISSUE THREAD

**Issue by jloh** [sveltia/sveltia-cms#267]
Hey all, thanks so much for creating Sveltia! I'm really enjoying it.

I'm trying to use the `datetime` widget but currently running into some issues with it dropping the timezone portion of a datetime string when saving/updating posts and having weird behaviour with the _Now_ button.

My Sveltia config for this collection looks like the below and I'm using Sveltia version `0.47.2` installed via NPM and manually initialised like this:

```javascript
window.CMS_MANUAL_INIT = true;
import CMS, { init } from "@sveltia/cms";
init();
```

Config:

```yaml
micro:
  name: "micro"
  label: "Micro"
  icon: note_stack
  folder: "content/micro"
  slug: "{{year}}-{{month}}-{{day}}-{{hour}}{{minute}}"
  summary: "{{date | date('YYYY-MM-DD h:mm a')}} - {{body | truncate(60, '&hellip;')}}"
  create: true
  fields:
    - { label: "Title", name: "title", widget: "datetime", default: "{{now}}", format: "YYYY-MM-DD HH:mm" }
    - {
        label: "Publish Date",
        name: "date",
        widget: "datetime",
        default: "{{now}}",
        format: "YYYY-MM-DDTHH:mm:ssZ",
      }
    - { label: "Tags", name: "tags", widget: "list", "default": [], label_singular: "Tag", required: false }
    - { label: "Draft", name: "draft", widget: "boolean", default: true }
    - { label: "Body", name: "body", widget: "markdown" }
```

This works _if_ I don't hit the _Now_ button on the widget. If I don't hit it, the `title` and `date` fields get rendered like this in the resulting document:

```yaml
title: 2024-12-05 00:21
date: 2024-12-05T00:21:00+11:00
```

If I hit _Now_ to update the timestamp they get rendered like this:

```yaml
title: 2024-12-05T00:17
date: 2024-12-05T00:17
```

Not quite sure if I'm doing something wrong there, I would expect them to follow the format set in my config.

I've also noticed if I update a post and _don't_ alter the timestamp it drops the timezone.

For example with the same config, if I have a document with a `date` field like this:

```yaml
date: "2024-12-04T07:31:06.000000Z"
```

Then save the document after editing the body (not touching the date fields), `date` gets updated to this:

```yaml
date: 2024-12-04T07:31:00+11:00
```

Which in this case since it moves from UTC to my current timezone means the posts time drastically changes.

Again, thanks for Sveltia. I really love it, the UI is nice and its really fast. Let me know what I can do to help debug.

**Comment by kyoshino** [comment]
Hey, thanks for the detailed bug report! I’m sure the DateTime widget has some issues 😅 Will check out today.

**Comment by kyoshino** [comment]
The Now button issue is solved with [v0.47.3](https://github.com/sveltia/sveltia-cms/releases/tag/v0.47.3) 🙇🏼 

As for the UTC issue, you need to use the `picker_utc` option for the [DateTime](https://decapcms.org/docs/widgets/#datetime) widget. If `true`, Sveltia CMS will save the value as:

```yaml
date: 2024-12-04T07:31:00+00:00
```

`Z` is replaced with an offset with UTC according to the [Moment.js doc](https://momentjs.com/docs/#/displaying/format/) and [Day.js doc](https://day.js.org/docs/en/display/format) (Sveltia CMS will switch from Moment.js from Day.js sometime soon to follow Decap CMS). If you want to keep the literal `Z`, you can change the `format` to `YYYY-MM-DDTHH:mm:ss[Z]` or simply omit the `format` option. 🙂

**Comment by jloh** [comment]
Hey, thanks so much for the fast fix! The Now button works perfectly 😄 

> As for the UTC issue, you need to use the `picker_utc` option for the [DateTime](https://decapcms.org/docs/widgets/#datetime) widget. If `true`, Sveltia CMS will save the value as:
> 
> ```yaml
> date: 2024-12-04T07:31:00+00:00
> ```
> 
> `Z` is replaced with an offset with UTC according to the [Moment.js doc](https://momentjs.com/docs/#/displaying/format/) and [Day.js doc](https://day.js.org/docs/en/display/format) (Sveltia CMS will switch from Moment.js from Day.js sometime soon to follow Decap CMS). If you want to keep the literal `Z`, you can change the `format` to `YYYY-MM-DDTHH:mm:ss[Z]` or simply omit the `format` option. 🙂

Just to make sure I'm understanding correctly, the fix for this is to set the date picker to use UTC? That way no matter what TZ I'm editing posts in it'll always use UTC? I'm not against that but the thing I'm struggling with is it not parsing the timezone and updating the time correctly.

For example if I were to move timezone and edit a post, it'd change the time without me knowing.

For example I just trialed editing a different post no longer in UTC:

```diff
-date: 2024-12-06T11:47:00+09:00
+date: 2024-12-06T11:47:00+11:00
```

And it resulted in that. Completely fine if the answer is just store it all in UTC for now to avoid this, just want to make sure I'm understanding correctly 😄

**Comment by kyoshino** [comment]
I think you’re understanding correctly 🙂 `picker_utc: true` makes the date picker and output to always use UTC.

> For example I just trialed editing a different post no longer in UTC:
> 
> ```diff
> -date: 2024-12-06T11:47:00+09:00
> +date: 2024-12-06T11:47:00+11:00
> ```

I think this is a bug. Will test and fix that.

**Comment by kyoshino** [comment]
The wrong time zone shifting issue is solved in [v0.47.6](https://github.com/sveltia/sveltia-cms/releases/tag/v0.47.6). Soon I’ll write full automated tests before migrating to Day.js.

**Comment by jloh** [comment]
Thanks! I can confirm its all working perfectly 😄 Thanks again for such fast fixes, I really appreciate it! Exicted to see where Sveltia goes, already loving it.
